File: data/EXP01/go/2257/2257.output_1.go
func countUnguarded(m int, n int, guards [][]int, walls [][]int) int {
    grid := make([][]int, m)
    for i := 0; i < m; i++ {
        grid[i] = make([]int, n)
    }
    
    for _, guard := range guards {
        row, col := guard[0], guard[1]
        grid[row][col] = -1
    }
    for _, wall := range walls {
        row, col := wall[0], wall[1]
        grid[row][col] = -2
    }
    
    unguarded := 0
    directions := [4][2]int{{-1, 0}, {1, 0}, {0, -1}, {0, 1}}
    
    var dfs func(row, col int)
    dfs = func(row, col int) {
        if row < 0 || row >= m || col < 0 || col >= n {
            return
        }
        if grid[row][col] == -1 || grid[row][col] == -2 {
            return
        }
        if grid[row][col] == 0 {
            unguarded++
        }
        grid[row][col] = -1 // mark as visited
        for _, direction := range directions {
            newRow, newCol := row+direction[0], col+direction[1]
            dfs(newRow, newCol)
        }
    }
    
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if grid[i][j] == 0 {
                dfs(i, j)
            }
        }
    }
    
    return unguarded
}

File: data/EXP01/python/2257/2257.output_1.py
# @lc app=leetcode id=2257 lang=python3
from typing import List

class Solution:
    def countUnguarded(self, m: int, n: int, guards: List[List[int]], walls: List[List[int]]) -> int:
        grid = [[0] * n for _ in range(m)]  # Initialize the grid with all cells set to 0
        
        for guard in guards:
            self.markGuardedCells(grid, m, n, guard)  # Mark cells seen by guards
            
        for wall in walls:
            self.markGuardedCells(grid, m, n, wall)  # Mark cells corresponding to walls
            
        count = 0
        
        for i in range(m):
            for j in range(n):
                if grid[i][j] == 0:  # Count unguarded cells
                    count += 1
                    
        return count
    
    def markGuardedCells(self, grid: List[List[int]], m: int, n: int, position: List[int]) -> None:
        directions = [(0, 1), (1, 0), (0, -1), (-1, 0)]  # Cardinal directions
        
        for direction in directions:
            row, col = position
            
            while 0 <= row < m and 0 <= col < n and grid[row][col] != 1:  # Move in the current direction until we reach a wall or another guard
                grid[row][col] = 1
                row += direction[0]
                col += direction[1]